integration_tests/auth/conftest.py

```diff
@@ -0,0 +1,91 @@
+import os
+from dataclasses import asdict
+from datetime import datetime
+from typing import Any, Callable, ContextManager, Dict, Iterator, List, Optional, Protocol, cast
+from unittest import mock
+from urllib.parse import urljoin
+
+import httpx
+import pytest
+from _pytest.fixtures import SubRequest
+from faker import Faker
+from phoenix.auth import REQUIREMENTS_FOR_PHOENIX_SECRET
+from phoenix.config import (
+    ENV_PHOENIX_ENABLE_AUTH,
+    ENV_PHOENIX_SECRET,
+    ENV_PHOENIX_SQL_DATABASE_URL,
+    get_base_url,
+)
+from sqlalchemy import URL, make_url
+from typing_extensions import TypeAlias
+
+ProjectName: TypeAlias = str
+Name: TypeAlias = str
+ApiKey: TypeAlias = str
+
+
+class GetGqlSpans(Protocol):
+    def __call__(self, *keys: str) -> Dict[ProjectName, List[Dict[str, Any]]]: ...
+
+
+@pytest.fixture(autouse=True, scope="module")
+def env_phoenix_enable_auth(fake: Faker) -> Iterator[None]:
+    secret = fake.unique.password(**asdict(REQUIREMENTS_FOR_PHOENIX_SECRET))
+    values = (
+        (ENV_PHOENIX_ENABLE_AUTH, "true"),
+        (ENV_PHOENIX_SECRET, secret),
+    )
+    with mock.patch.dict(os.environ, values):
```

Comment:
Suggestion: `monkeypatch.setenv`

Reply:
`monkeypatch` is function scope. This is class scope